sdk/core/azure_core/src/http/poller.rs

```diff
@@ -600,77 +640,160 @@ where
                 Ok(PollerResult::Done { response }) => (Ok(response), State::Done),
                 Ok(PollerResult::Succeeded {
                     response,
                     target: get_target,
                 }) => {
                     // Send the target callback through the channel
-                    if let Some(tx) = target_tx {
-                        let _ = tx.send(get_target());
+                    if let Some(tx) = poller_stream_state.target_tx.take() {
+                        let _ = tx.send((
+                            get_target(),
+                            if poller_stream_state.added_span {
+                                Some(poller_stream_state.options.context.clone())
+                            } else {
+                                None
+                            },
+                        ));
                     }
-                    // Also yield the final status response and set target_tx to None since we consumed it
-                    return Some((Ok(response), (State::Done, make_request, None)));
+                    // Also yield the final status response
+                    poller_stream_state.state = State::Done;
+                    return Some((Ok(response), poller_stream_state));
                 }
             };
 
-            // Flow 'make_request' and target_tx through to avoid cloning
-            Some((item, (next_state, make_request, target_tx)))
+            // Update state and return
+            poller_stream_state.state = next_state;
+            Some((item, poller_stream_state))
         },
     );
 
     let target = Box::new(async move {
         match target_rx.await {
-            Ok(fut) => fut.await,
+            Ok(target_state) => {
+                // Await the target future to get the final response from the poller.
+                let res = target_state.0.await;
+                // If we added a span to the target, take the result of the final target future to finalize the span.
+
+                if let Some(ctx) = target_state.1 {
+                    match &res {
+                        Ok(response) => {
+                            // When the result is done, finalize the span. Note that we only do that if we created the span in the first place,
+                            // otherwise it is the responsibility of the caller to end their span.
+                            if let Some(span) = ctx.value::<Arc<dyn Span>>() {
+                                // 5xx status codes SHOULD set status to Error.
+                                // The description should not be set because it can be inferred from "http.response.status_code".
+                                if response.status().is_server_error() {
+                                    span.set_status(SpanStatus::Error {
+                                        description: "".to_string(),
```

Comment:
Shouldn't we at least call `to_string()` on the `StatusCode`? Ideally, we use the "code" embedded in our error since that logic is already the Azure standard error description or falls back to the stringified status code.